vigranumpy/test/test_segmentation.py

```diff
@@ -33,6 +33,49 @@ def test_labelMultiArray():
     _impl_test_labelMultiArray(numpy.uint8)
     _impl_test_labelMultiArray(numpy.uint32)
     _impl_test_labelMultiArray(numpy.float32)
+
+
+def _impl_test_applyMapping(dtype):
+    original = numpy.arange(100, dtype=dtype ).reshape(10,10)
+    mapping = { int(k) : int(k + 100) for k in original.flat[:] }
+
```

Comment:
You know a lot more about `boost::python` than I do, so I might be missing something.  But I don't think we need to implement a replacement for `extract<>()`, I think we just need to register the right conversion functions with `boost::python` so that `extract<>()` works correctly for `np.uintXX` objects.

I've implemented a prototype that illustrates the concept here:
https://github.com/stuarteberg/printnum/blob/master/src/printnum.cpp

That code works, but it probably has **terrible** performance. It calls a bajillion python functions just to convert a simple integer from one type to another.  If I knew the proper incantations for doing the same thing using only the Python C-API, then that would be a faster way to implement this code.

For the record, the output of `test.py` in that repo is this:

```
$ PYTHONPATH=build python test.py
The number is: 123
The other number is: -456
```